### ISSUE TITLE
slight copyright issues

### ISSUE THREAD

**Issue by maxvollmer** [DrBeef/Lambda1VR#17]
Hi!

I already sent you an email, but I also post here because I feel this is important.

1. You cannot put this code under the GPL. That violates the Valve SDK license, which covers parts of the code you are sharing here. Since GPL doesn't allow you to mix GPL licensed code with non-GPL licensed code, I recommend using MIT for the non-Valve-SDK-license parts and keeping the Valve SDK license for the applicable parts.

2. You are using my code for your mod (see here: https://github.com/maxvollmer/Half-Life-VR). My license is very lenient and allows you to do so - as long as you give credit. While admittedly you have a tiny mention of my name at the end of your readme, it doesn't really reflect which parts in your source repo are from me. I am also unhappy that you didn't link back to my original work. And, in part for the reasons mentioned in the former point, I do not allow my code to be put under a GPL license. Please update this.

3. You are using formic's weapon models that he made for my Half-Life: VR mod. He did not give you permission to do so. That is between you two, but I seriously suggest you ask people for permission before using their assets. Especially when adding a paypal donate button on the website where you publish other people's work without proper credit, or lack of permission.

I don't want you to shut down this project, I think it's cool you made this, but please be a bit more proactive in your communication and make sure you don't accidentally steal other people's work.

I kindly ask you to:
a) contact formic and apologize and ask for permission to use his models. I know he is a bit unhappy right now, but he is a reasonable and nice person, so I am sure you can work this out.
b) credit me properly here and on your website.
c) add a link to my github and patreon on your website at least as prominent as your own paypal donate button.

And as also said in my email: If you want to collaborate, reach out!

Thanks!

**Comment by DrBeef** [comment]
Dear Max

Thank you for your concern. You have raised some large issues here publicly that I therefore have to justify and respond to. I had received your e-mail last night, and was preparing a thought out response. Since you have now elevated that conversation into the public sphere I think it best I respond here in similar fashion. 

1)	You have made the accusation here and in your e-mail that I took your work, improperly accredited, and that the permission on your project meant that you “did not mean publish this as your own work with a link to your Paypal and put my name somewhere in the fine print.” (from your e-mail)

I completely refute the fact that I took this initial work and claimed it as my own. 

I have used some of your 6DOF code. It is worth mentioning that I had to modify significantly to get it to work within my port. It is also worth mentioning that this element represents a very small fraction of the overall work I have had to do to bring this to Oculus Quest. I appreciate that you feel aggrieved but the inference that this was some cash-grab of your work is both completely incorrect and personally insulting. 

I have put the accreditation in the credits section of both my Github and Website with the same visibility as Xash3D (the actual engine required to run Half Life on Android), GLES4ES – the mod that actually made OpenGL 2.0 calls possible on legacy systems and SideQuest. I do not believe this constitutes “in the small print”.

2. There was no specific license on your project at all, just a "feel free to use this and give credit". I did exactly that. The website is being updated to include a link to your github project that the 6DoF weapon code came from. There is no specific mention of the particular type of license and that it can't be put under GPL. Here's a screenshot of the license terms from the readme:

![image](https://user-images.githubusercontent.com/6168240/62278850-ceaf1c80-b440-11e9-9e31-4dcfcd19e94b.png)

Also, only the weapon logic is from your publicly available code, HMD positional tracking is my own scheme, the Crowbar swinging is probably the lion's share of the functionality, I will make that clearer.

3. I did email formic to ask for permission to use his models well before this was released (see screenshot of that email below). I received no reply. I didn't know what else to do at that point and as they were in the public domain I included them. If he is upset about this then I am very sorry but as I have no other way of contacting him than the email I used already I am not sure what to do about it. To be fair, I did also credit him with the weapon models in the same way I credited everyone else:

![image](https://user-images.githubusercontent.com/6168240/62278083-4aa86500-b43f-11e9-906f-b0d8e163f238.png)

If he wants to drop me a line I would be happy to chat to him and apologise for the miscommunication!

I have a credits list, it isn't hidden, it is clear for all to read. I am happy to credit you and link to the GitHub (which I have now done). There is no legal requirement for me to include your Patreon.

I fell that saying the following was unnecessary:

>> I don't want you to shut down this project

All that said, I would be happy to consider collaboration in the future, but at the moment I am feeling like I am being excessively accused of actions that are simply misunderstandings.

Regards
Simon

**Comment by baggyg** [comment]
I just thought I would chime in here regarding the concern Max had over software licensing. I am not a software licensing expert but just trying to help find the right solution if there is an underlying problem. Licensing can be difficult to get right, especially for modifications. I think an issue, if there is one at all, arises because this project's code is based entirely on the Xash3D build from https://github.com/FWGS/xash3d. That project is entirely GPL3.0 licensed so it was appropriate (and indeed necessary) to retain the same license for this project. The SDK files within the project used by Xash contain their own header with licensing terms. My interpretation here is that the wrapper around that library is GPL3.0 with those files having a different license. Again this has been retained.

**Comment by cercata** [comment]
> https://github.com/FWGS/xash3d. That project is entirely GPL3.0 licensed so it was appropriate (and indeed necessary) to retain the same license for this project. The SDK files within the project used by Xash contain their own header with licensing terms. My interpretation here is that the wrapper around that library is GPL3.0 with those files having a different license. Again this has been retained.

We would need an army of lawyer to investigate all the posible ramifications of the GPL 3.0 ... but in this case it seems that if there is a problem, is xash3d who should change it ...

**Comment by neitey** [comment]
> 2. You are using my code for your mod (see here: https://github.com/maxvollmer/Half-Life-VR). My license is very lenient and allows you to do so - as long as you give credit. While admittedly you have a tiny mention of my name at the end of your readme, it doesn't really reflect which parts in your source repo are from me. I am also unhappy that you didn't link back to my original work. And, in part for the reasons mentioned in the former point, I do not allow my code to be put under a GPL license. Please update this.

As far as i can tell the name Max Vollmer was credited right from the start on the official website www.lambda1vr.com/, in the same font size as all the other content. So wouldn't say it was in any way fine printed or tiny. So for me as an outsider, it was pretty clear that the initial 6DoF code came from Vollmer. I hope you guys can work it out.

**Comment by cercata** [comment]
> All that said, I would be happy to consider collaboration in the future

I have seen many people asking if this is gonna be ported to PCVR ... it would be nice it maxvollmer could benefit from this work to finish this PC

**Comment by formicsapien** [comment]
@DrBeef You did email me.  I did not see the email until I checked my email after reading your response here.  I'm working on a response now...

**Comment by formicsapien** [comment]
@cercata Our team has been working under Max over the last two years, and we've made a lot of progress!  We're not quite ready for a public release, but we will be soon.  That's the reason Max reached out to DrBeef in the first place here!  It would've been awesome if we could work together., and I'm baffled as to why DrBeef responded the way he did.  Max reached out a hand for cooperation.

**Comment by a1batross** [comment]
Why you asking Simon, instead of creating such issue in FWGS repository or asking Unkle Mike directly?

You can't just go and relicense Xash under any other license that you want to, without asking permission from FWGS, Unkle Mike and even id Software.

At first, GPL allows you to work with proprietary code, when it's you use proprietary code in your FLOSS project, but not otherwise, when proprietary devs are using your free library(and that's why LGPL exists). But it's not a GPL violation when GPL licensed program works in Windows, for example, where it needs to be linked with Microsoft's proprietary code. 

**BUT**: It's only in dynamic linking. There is no static linking. For static linking it's completely prohibited to mix free and proprietary code (without providing object files for LGPL).


I know how engine works, no one forces you to use HLSDK libraries. You can just write your own, in GPL, build it and run with same engine binary without recompiling.

At second, HLSDK is licensed under custom license, you can actually read in LICENSE file where it states that you can use it ONLY in Half-Life Engine. HL Engine is only GoldSource, it's proprieraty. Even if you reverse engineer it, it would not count as "HL Engine".

So, you probably get it, there is no legal way to create a wrapper or like that, just to comply GPL. It's just a minor issue compared to HLSDK.

It's completely useless to take your time to make wrappers and other stuff that would comply the GPL and HLSDK licenses both. The only real solution is to rewrite Half-Life both client and server library under other license, maybe better in other langauge and by someone who've never seen HLSDK code, i.e. clean room. 

Would you take your time to make it, when Valve don't cares for such long time(Xash exists for more than 10 years already)? I'm not.

I'm thankful to Valve, for not being Nin****do or Bet****da.

**Comment by formicsapien** [comment]
@neitey There was no mention of nor link for Max Vollmer on the website until after this issue was opened.

**Comment by a1batross** [comment]
Ok. I need to sleep more. 

I've just realized that Max is talking about his own code in HLSDK parts. Oh well.

**Comment by cercata** [comment]
@formicsapien that's cool you have been working on the project.
Personally I thought it was dead, I had checked the repo several times, and not a single update ...
Now I see there are private updates on Patreon. It wouldn't be bad to update at least the readme on github with a line ;)

**Comment by cercata** [comment]
> Would you take your time to make it, when Valve don't cares for such long time(Xash exists for more than 10 years already)? I'm not.
> 
> I'm thankful to Valve, for not being `Nin****do` or `Bet****da`.

Yeah, Gabe is a smart man, like Stallman, they put licences to avoid big corporartions profiting from free software without adding value to it, not to engage fights between talented modders ;)

**Comment by maxvollmer** [comment]
Okay, so first of all I want to apologize that my original post sounded accusatory! I admire your work and I definitely do not think anything mentioned happened in bad faith.

I agree there are a lot of misunderstandings here, and I am sure we can resolve it all. I really hope in a year's time we can all laugh about this: "hey you remember how I got onto you like an arse?" "yeah and i got all defensive and whatnot" "oh the good times" ;)

I understand now that you only took parts of my original code. I would ask you to maybe rephrase the credits for my code as "Max Vollmer for their 2017 Half-Life VR mod, of which I used and mofidied the 6DoF weapons code for this mod." or similar. I honestly simply misunderstood your project as a port of my mod to the Quest. I am sorry.

As for the licensing issues. This is still a problem, because my code is based on Valve's Half-life SDK. This cannot be mixed with GPL licensed code and cannot be relicensed under GPL (you're basically open sourcing and releasing Valve's IP). I strongly recommend you split your project: The xash based engine part, and the Half-Life SDK based client.dll and hl.dll, which can contain my code. The latter has to remain under the "VALVE SDK LICENSE". Then you should be good legally. I mean, Valve is super chill, but still, it's not that much work to do it right.

As for formic's weapons. I am sorry he didn't see your email, but his weapons are not public domain. They can't be, as they also are part of the Valve SDK and are Valve's IP and formic's work. If an author doesn't respond to your request, that doesn't give you a free pass to use the work. Again, I honestly think this was a simple mistake, not in any way you acting in bad faith, and I am sure now that we talked this out formic will be happy to officially give you permission to use the weapons for your mod.

Also of course there is no legal obligation to link to my patreon, I am sorry that I overstepped in my request/demands.

**Comment by DrBeef** [comment]
Hi Max

Thank-you very much for the above response, it is very much appreciated indeed, I think we have definitely got off on the wrong foot! There are definitely some misunderstandings here and with any luck we'll sort them out and be able to work together / joke about in the future.

Firstly, apologies I hadn't included the link to your GitHub in the first release of the website credits as @formicsapien pointed out, I am very happy to rephrase the credits accordingly too. I would certainly be well behind where I am on development (in particular the crowbar!) were it not for the changes you made public, so I am obviously very grateful for them.

On the licensing side I completely see your point.. I realise now that the Xash3D engine is separate to the hlsdk in exactly the way you describe, I mistakenly committed the whole lot together which is not the right way to do it. I would obviously like to be doing things the correct/legal way, especially as some of your code is also included in there too, so I will look at splitting the client/HLSDK portions off into a separately licensed repo that contains your original changes all under the appropriate license in the same way Xash3D do it. I agree, hopefully it shouldn't take too much to make those changes I think, but please bear with me while I make them. If, once it is done, it's still not quite structured correctly please let me know.

@formicsapien , my sincerest apologies for using your weapon models without getting actual permission. The reason it played out like it did was because at the time I saw them in the https://github.com/rorincloud/Half-Life-VR repo, which appeared to be abandoned (2 years since last commit) and as I received no reply to the email I naively (and wrongly) assumed they were long since forgotten and left in the public domain, with hindsight I should have created a GitHub issue to contact you. I actually had no idea that you were working with @maxvollmer on his mod, had I done so I'd have been in touch with you guys much sooner and none of this would've even happened :(
Unless you give me express permission to use your weapon models (and I understand if you would rather I didn't because I imagine they form part of the work you are working on with Max) then I will remove them from my mod. If you _are_ happy for me to continue using the models then I will obviously be incredibly grateful, but please feel under no obligation to do so.

I think, with my above changes to bit n' pieces we'll all hopefully be happy and as you say, will just have a bit of a laugh about it in the future. I'll postpone further updates/releases until all's settled.

Feel free to correspond further directly by email. I would obviously be very keen to hear about some of the other changes you are making for your Mod!

Best regards
Simon

**Comment by maxvollmer** [comment]
Closing this, as I think all has been said and things are cleared up now :)

Also I am in London in September. Let's meet for a snack and a chat!
